CDTDatastore/HTTP/CDTHTTPInterceptorContext.h

```diff
@@ -39,7 +48,8 @@ NS_ASSUME_NONNULL_BEGIN
  *
  *  @param request the request this context should represent
  **/
-- (instancetype)initWithRequest:(NSMutableURLRequest *)request NS_DESIGNATED_INITIALIZER;
+- (instancetype)initWithRequest:(NSMutableURLRequest *)request
+                          state:(NSMutableDictionary *)state NS_DESIGNATED_INITIALIZER;
```

Comment:
I don't think we can just add the `NSMutableDictionary` to the init method, since the original API was envisioned as a user being able to return their own context with any customisation they want, so this would be a breaking API change. Basically this comes down to whether this is deemed API or not, my stance on that from thinking back is that it is.